server/modules/poem_bot.ts

```diff
@@ -0,0 +1,140 @@
+import { ChatOpenAI } from "@langchain/openai";
+import { InMemoryChatMessageHistory } from "@langchain/core/chat_history";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+import { RunnableWithMessageHistory } from "@langchain/core/runnables";
+import * as dotenv from "dotenv";
+import { botDeviceIDToMessageHistory } from "./globals";
+import { StringOutputParser } from "@langchain/core/output_parsers";
+import { analyzeSystemPrompt, analyzeUserPrompt, completeSystemPrompt, completeUserPrompt } from "../llm_utils/prompts";
+import { IGameSettingsInfo } from "../../src/types";
+import { lineConstraints } from "../../src/constants";
+import { canBeFixedByShifting, isAcceptableShape, processPoetryLines } from "../llm_utils/llm_funcs";
+import { logger } from "../utilities/loggerUtils";
+
+dotenv.config({ path: __dirname + "/../.env" });
```

Comment:
We could skip this in this MR but would be nice to clean up the extra dotenv imports and just let the base server file import it once